pkg/controller/manager/manager_controller.go

```diff
@@ -495,6 +505,14 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 	// Create a component handler to manage the rendered component.
 	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
 
+	// Reset replicas to 1 in management or managed clusters
+	// TODO: Support MCM tigera-manager HA deployment
+	if managementCluster != nil || managementClusterConnection != nil {
+		if *instance.Spec.Replicas != 1 {
+			instance.Spec.Replicas = ptr.Int32ToPtr(1)
```

Comment:
We shouldn't override this in the case of a managementCluster but should be verifying that with a managementCluster the field is set to 1.
Especially since 1 is the default value, which would mean that a user would explicitly be setting the value to 2.

Reply:
@tmjd Just want to make sure I understand your comments correctly. This check will be removed when MCM HA is supported. You are suggesting to check for replicas but not overriding the user settings? For management and manged clusters, if a user sets it to be not 1, shall we set a degraded state in the Manager component?

Reply:
I think because we changed the API and aren't providing an individual knob for manager replicas, I think the behavior of forcing to 1 replica for MCM clusters is acceptable.

Reply:
Probably OK to default it, but we still need to respect the value the user provided if its set (or error)

Reply:
I was expecting to include a comment on the field in Installation that it doesn't apply to the manager when using multi-cluster management. If that is not ok, we'll add a change in the cloud build to ignore the setting for the manager so we can utilize HA for the other components.

Reply:
not a fan of ignoring user intent in the general case. I don't think most people will want to set `replicas: 2` only to have it ignored, even if running a management cluster.

If we need, we can either have the cloud build do something special, or introduce a new replicas field that applies only to the manager component.

Reply:
What is the concern if we lock the manager UI to 1 in the case of MCM. We're not removing previous available functionality but allowing this feature to be used with compatible components until the manager can also support HA with MCM.
It feels kind of the same as saying we can't do Replicas because we can't run all our components with multiple replicas.
But I do understand that it is not as straight forward since it would be necessary to include that caveat.

Reply:
I suppose so long as we're comfortable with the following things it's OK:
- Bug reports of "I specified X replicas but still only have 1"
- Changing the behavior of the `ControlPlaneReplicas` field to apply to a different set of components in the future, and any ensuing bug reports.

Agree that they are probably pretty minor, but a strict interpretation of the API "rules" would attempt to avoid those.

Reply:
I suppose if we say this field applies to all control plane components that support HA, then we're OK. That sort of needs to be the case anyway, because kube-controllers.

Reply:
Update the description for `ControlPlaneReplicas` to the following:

```golang
// Replicas defines how many replicas of the control plane core components will be deployed.
// This field applies to all control plane components that support High Availability. Defaults to 2.
// +optional
```